### PR TITLE
(SIMP-3473) Address comments for simp_snmpd profile module

### PR DESCRIPTION
**Pull request by jeannegreulich**
- fixed agentaddress and firewall
  - remove junk files and fixed typos
  - added settings for agent
  - updated the readme

(SIMP-3473) #close

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -10,11 +13,64 @@
...
 #   The version of snmp protocol to use.
-#   At this time the simp_profile only manages v3, to configure
+#   At this time the simp_snmpd profile only manages v3, to configure
 #   older versions use the snmp module directly.
+# @param  $snmp_conf_file,
+#   A file of snmp.conf directives that is included for configuration directives.
+#   this file is managed by puppet.
+# @param  $simp_snmpd_dir,
+#   a directory of *.conf files which include snmpd directives.  Files in this
+#   directory are managed by puppet.
```
Comment (lnemsick-simp) [inline manifests/init.pp]: Is the directory managed?  In other words, will files not under puppet control be removed from this directory?

File: manifests/init.pp [hunk 2]
```diff
@@ -10,11 +13,64 @@
 #   if false packages will be installed with "present"
 # @param  $version
```
Comment (lnemsick-simp) [inline manifests/init.pp]: $version should be version. puppet strings fails when you have the $ 

Note that `puppet strings generate` spews many warnings that need to be fixed!

File: manifests/init.pp [hunk 3]
```diff
@@ -10,11 +13,64 @@
...
+#
+# SNMPD Agent Parameters
+# @param agentaddress
+# @see man snmpd  in the LISTENING ADDRESSES section for more details.
+#   An array of listening addresses for the snmpd to listen on. Each
+#   element of the array is an array of the form.  This array is
+#   also used by the config/firewall.pp to open ports if iptables
+#   is being used.
+#   following:
+# @params $do_not_log_tcpwrappers
```
Comment (lnemsick-simp) [inline manifests/init.pp]: @params is not a valid tag for puppet strings

File: README.md [hunk 4]
```diff
@@ -20,86 +15,148 @@
...
+
+`simp_snmpd` configures some basic system information: contact, location,
+system name, and services, in the snmpd configuration directory.  These settings
+can be changed via hiera, instantiation, by creating your own configuration file
+in the user directory.
+
+NOTE: If the system variables are set in a configuratioin file then net-snmp
+marks them as not writable and will not allow them to be changed via snmpset
+or other client utilities.
+To be able to set information via a client, set `simp_snmpd::system_info` to
```
Comment (lnemsick-simp) [inline README.md]: back quotes around snmpset

File: manifests/init.pp [hunk 5]
```diff
@@ -10,11 +13,64 @@
...
+# @param   $snmpd_service_ensure, $trap_service_ensure
+#   Set the snmpd/trap daemon service to stopped or running
+# @param $snmpd_service_startatboot, $trap_service_startatboot
+#   Start the snmpd/trap service at boot
+#
+# SNMPD Agent Parameters
+# @param agentaddress
+# @see man snmpd  in the LISTENING ADDRESSES section for more details.
+#   An array of listening addresses for the snmpd to listen on. Each
+#   element of the array is an array of the form.  This array is
```
Comment (lnemsick-simp) [inline manifests/init.pp]: What does "Each element of the array is an array of the form.' mean?

File: manifests/init.pp [hunk 6]
```diff
@@ -10,11 +13,64 @@
...
+#   Whether to enable rsync for MIBS
+# @param $rsync_mibs_dir
+#  The full path for the directory to rsync mibs too.  It does not
+#  remove what is already there.
+#
+# USM/VACM parameters
+# @params v3_users_hash
+# @see man snmpd.conf  SNMPv3 with the User-based Security Model (USM) section
+#   A hash of users to create for usm access. Also see README for details
 # @param $v3_users_hash,  $view_hash, $group_hash, $access_hash,
```
Comment (lnemsick-simp) [inline manifests/init.pp]: Need individual @param line for each parameter

File: manifests/init.pp [hunk 7]
```diff
@@ -26,63 +82,125 @@
...
-#   The addresses to have the snmpd listen on.
-
+# snmp.conf access configuration default items
+# @param  $defauthtype,
+#   The default authentication type used for clients.
+# @param  $defprivtype,
+#   The default privacy type used for encrypting communication when using usm.
+# @param  $defsecuritymodel,
+#   currently simp_snmpd only supports the usm security model it will support
+#   tsm  in the near future.  This option determins if usm or tsm access is
```
Comment (lnemsick-simp) [inline manifests/init.pp]: - Is tsm now possible?  I see a tsm.pp.
- 'determins' -> 'determines'

File: manifests/init.pp [hunk 8]
```diff
@@ -26,63 +82,125 @@
...
+#   snmp messages to /var/log/snmpd.log and set up log rotation.
+# @params $tcpwrappers
+#   Whether or not the system is using tcpwrappers to control access.
+#
 class simp_snmpd (
   Enum['present','absent']             $ensure,
-  Integer[3,3]                         $version,
-  #  Boolean                           $snmptrapd, Just set to false in the snmp call.
-  #  Boolean                           $snmpd,   Just set to true in the snmp call.
+  Integer                              $version,
```
Comment (lnemsick-simp) [inline manifests/init.pp]: Why was the version pinning to 3 replaced with error messages below?

File: templates/snmpd/snmpd.conf.erb [hunk 9]
```diff
@@ -5,23 +5,23 @@
 ################################################################################
 # AGENT BEHAVIOUR
 
-<% if @master != nil -%>
+<% if @master.nil? -%>
```
Comment (lnemsick-simp) [inline templates/snmpd/snmpd.conf.erb]: Is this change from checking if not nil to checking if nil correct?  Did you mean to check 
   unless @master.nil?

Numerous other of these types of check problems.  Would be caught by a thorough unit test

File: templates/snmpd/snmpd.conf.erb [hunk 10]
```diff
@@ -5,23 +5,23 @@
...
 <% if @master -%>
```
Comment (lnemsick-simp) [inline templates/snmpd/snmpd.conf.erb]: We have a convention of indentations that make the ERB code easier to read:
<% if condition -%>
<%   if nested_condition -%>
<%   end -%>
<% end -%>

File: templates/snmpd/snmpd.conf.erb [hunk 11]
```diff
@@ -5,23 +5,23 @@
 ################################################################################
```
Comment (lnemsick-simp) [inline templates/snmpd/snmpd.conf.erb]: Why do we have a mix of erb and epp files?  Seems like we should be one or the other.

File: templates/snmpd/snmpd.conf.erb [hunk 12]
```diff
@@ -5,23 +5,23 @@
...
 # AGENT BEHAVIOUR
 
-<% if @master != nil -%>
+<% if @master.nil? -%>
 <% if @master -%>
 master agentx
 
-<% if @agentx_perms != nil -%>
+<% if @agentx_perms.nil? -%>
```
Comment (lnemsick-simp) [inline templates/snmpd/snmpd.conf.erb]: Should be
   unless @agentx_perms.nil?

File: templates/snmpd/snmpd.conf.erb [hunk 13]
```diff
@@ -5,23 +5,23 @@
...
 agentXPerms <%= @agentx_perms %>
 <% end -%>
-<% if @agentx_ping_interval != nil -%>
+<% if @agentx_ping_interval.nil? -%>
```
Comment (lnemsick-simp) [inline templates/snmpd/snmpd.conf.erb]: unless @agentx_ping_interval.nil?

File: templates/snmpd/snmpd.conf.erb [hunk 14]
```diff
@@ -5,23 +5,23 @@
...
 agentXPingInterval <%= @agentx_ping_interval %>
 <% end -%>
-<% if @agentx_socket != nil -%>
+<% if @agentx_socket.nil? -%>
```
Comment (lnemsick-simp) [inline templates/snmpd/snmpd.conf.erb]: unless @agentx_socket.nil?

File: templates/snmpd/snmpd.conf.erb [hunk 15]
```diff
@@ -5,23 +5,23 @@
...
 agentXSocket <%= @agentx_socket %>
 <% end -%>
-<% if @agentx_timeout != nil -%>
+<% if @agentx_timeout.nil? -%>
```
Comment (lnemsick-simp) [inline templates/snmpd/snmpd.conf.erb]: unless @agentx_timeout.nil?

File: templates/snmpd/snmpd.conf.erb [hunk 16]
```diff
@@ -5,23 +5,23 @@
...
 agentXTimeout <%= @agentx_timeout %>
 <% end -%>
-<% if @agentx_retries != nil -%>
+<% if @agentx_retries.nil? -%>
```
Comment (lnemsick-simp) [inline templates/snmpd/snmpd.conf.erb]: unless @agentx_retries.nil?

File: types/listeningaddr.pp [hunk 17]
```diff
@@ -1 +1 @@
-type Simp_snmpd::Listeningaddr = Tuple[Simp_snmpd::Protocol, Variant[Simplib::IP::V4, Simplib::IP::V6::Bracketed, Simplib::Hostname], Simplib::Port]
+type Simp_snmpd::Listeningaddr = Tuple[Simp_snmpd::Protocol, Variant[Simplib::IP::V4, Simplib::IP::V6::Bracketed, Simplib::Hostname, Stdlib::Unixpath], Optional[Variant[Simplib::Port,String]]]
```
Comment (lnemsick-simp) [inline types/listeningaddr.pp]: What is the purpose of the Optional[Variant[Simplib::Port,String]] ?
Do you have a test for this type?

File: types/timeout.pp [hunk 18]
```diff
@@ -0,0 +1 @@
+type Simp_snmpd::Timeout = Pattern[/^\d*[smhdw]?$/]
```
Comment (lnemsick-simp) [inline types/timeout.pp]: Do you have a test for this type?

File: lib/facter/net_snmpd_version.rb [hunk 19]
```diff
@@ -0,0 +1,27 @@
+# Returns the version of the snmpd daemon
```
Comment (lnemsick-simp) [inline lib/facter/net_snmpd_version.rb]: Do you have a test for this fact?

File: lib/puppet/functions/simp_snmpd/authaccesslist.rb [hunk 20]
```diff
@@ -10,7 +10,7 @@ def createlist(access_hash,defaultmodel,defaultlevel)
     access_hash.each { | name, values|
```
Comment (lnemsick-simp) [inline lib/puppet/functions/simp_snmpd/authaccesslist.rb]: Do you have a test for this function?

File: templates/snmpd/snmpd.conf.erb (hunk 12 above)
Comment (jeannegreulich) [inline templates/snmpd/snmpd.conf.erb]: This template is used by snmp module.  I just removed some things from it that I put other places.

**Pull request merged**